### PR TITLE
toggleLoading: fix race

### PR DESCRIPTION
**Pull request by samouri**
Fixes https://github.com/ampproject/amphtml/issues/27902

Interestingly enough, the existing test was just 1 line out-of-order to be able to catch this. This addressed a potential race condition between `viewportCallback` and `layoutCallback`. It is possible for layout to complete in less than 100ms, which would end up meaning that we call toggleLoading in the wrong order (`toggleLoading(false); toggleLoading(true);`)

### CODE REVIEW

File: test/unit/test-custom-element.js [hunk 1]
```diff
@@ -1995,8 +1995,8 @@ describes.realWin('CustomElement', {amp: true}, (env) => {
       it('should NOT turn on when enters viewport but already laid out', () => {
         stubInA4A(false);
         const toggle = env.sandbox.spy(element, 'toggleLoading');
-        element.layoutCount_ = 1;
         element.viewportCallback(true);
+        element.layoutCount_ = 1;
```
Comment (choumx) [inline test/unit/test-custom-element.js:1999]: This should probably be its own unit test (one for each conditional).

The outer conditional check should check that both `toggleLoading` and `timer.delay` are not called.

Comment (samouri) [reply test/unit/test-custom-element.js:1999]: I'll add the test and the submit when its green.

Comment (samouri) [reply test/unit/test-custom-element.js:2012]: Sure. Added

**Comment by samouri** [comment]
Through a terrifying sequence of events, I've somehow accidentally included this fix in https://github.com/ampproject/amphtml/pull/27834. I've even looked through my bash history and am unsure how this could have happened.

**Pull request merged**